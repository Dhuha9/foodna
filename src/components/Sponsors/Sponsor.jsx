import React from 'react';
import { Col } from 'react-bootstrap';
import PropTypes from 'prop-types';

function Sponsor({ websiteUrl, imageLogo, name }) {
  return (
    <Col>
      <a href={websiteUrl}>
        <img className="sponsor-image mx-4" src={imageLogo} alt={name}></img>
      </a>
    </Col>
  );
}

export default Sponsor;
Sponsor.propTypes = {
  websiteUrl: PropTypes.string,
  imageLogo: PropTypes.string,
  name: PropTypes.string,
};
