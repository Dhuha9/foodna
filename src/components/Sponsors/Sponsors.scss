.container {
  width: 830px;
}

.sponsor-image {
  width: 140px;
  height: 140px;
}

.heading {
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 140.62%;
  /* identical to box height, or 34px */
  text-align: center;
}

.subheading {
  left: 28.54%;
  right: 28.47%;
  top: 28.76%;
  bottom: 63.59%;

  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 20px;
  line-height: 28px;
  /* identical to box height */

  text-align: center;
}
