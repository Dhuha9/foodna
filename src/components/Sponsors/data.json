[
  {
    "name": "Re:coded",
    "websiteUrl": "https://www.re-coded.com/",
    "imageLogo": "recodedLogo"
  },
  {
    "name": "Re:coded",
    "websiteUrl": "https://www.re-coded.com/",
    "imageLogo": "recodedLogo"
  },
  {
    "name": "Re:coded",
    "websiteUrl": "https://www.re-coded.com/",
    "imageLogo": "recodedLogo"
  },
  {
    "name": "Re:coded",
    "websiteUrl": "https://www.re-coded.com/",
    "imageLogo": "recodedLogo"
  },
  {
    "name": "Re:coded",
    "websiteUrl": "https://www.re-coded.com/",
    "imageLogo": "recodedLogo"
  }
]
