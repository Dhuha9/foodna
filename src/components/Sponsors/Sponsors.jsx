import React from 'react';
import { Container, Row } from 'react-bootstrap';
import './Sponsors.scss';
import Sponsor from './Sponsor';
import data from './data.json';
import recodedLogo from '../../dist/img/ReCodedLogo.jpg';

const imgMap = { recodedLogo: recodedLogo };

function Sponsors() {
  return (
    <div>
      <h1 className="heading mb-4">Sponsors</h1>
      <p className="subheading mb-4">
        This project is generously sponsored by:
      </p>
      <Container>
        <Row className="justify-content-center" noGutters={true}>
          {data.map((sponsorsData, index) => (
            <Sponsor
              {...sponsorsData}
              key={index}
              imageLogo={imgMap[sponsorsData.imageLogo]}
            />
          ))}
        </Row>
      </Container>
    </div>
  );
}

export default Sponsors;
