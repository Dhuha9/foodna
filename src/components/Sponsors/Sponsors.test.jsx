import React from 'react';
import renderer from 'react-test-renderer';
import Sponsors from './Sponsors';

it('Sponsors snapshot', () => {
  const tree = renderer.create(<Sponsors />).toJSON();
  expect(tree).toMatchSnapshot();
});
