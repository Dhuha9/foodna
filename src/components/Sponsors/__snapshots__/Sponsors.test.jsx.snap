// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors snapshot 1`] = `
<div>
  <h1
    className="heading mb-4"
  >
    Sponsors
  </h1>
  <p
    className="subheading mb-4"
  >
    This project is generously sponsored by:
  </p>
  <div
    className="container"
  >
    <div
      className="justify-content-center row no-gutters"
    >
      <div
        className="col"
      >
        <a
          href="https://www.re-coded.com/"
        >
          <img
            alt="Re:coded"
            className="sponsor-image"
            src="ReCodedLogo.jpg"
          />
        </a>
      </div>
      <div
        className="col"
      >
        <a
          href="https://www.re-coded.com/"
        >
          <img
            alt="Re:coded"
            className="sponsor-image"
            src="ReCodedLogo.jpg"
          />
        </a>
      </div>
      <div
        className="col"
      >
        <a
          href="https://www.re-coded.com/"
        >
          <img
            alt="Re:coded"
            className="sponsor-image"
            src="ReCodedLogo.jpg"
          />
        </a>
      </div>
      <div
        className="col"
      >
        <a
          href="https://www.re-coded.com/"
        >
          <img
            alt="Re:coded"
            className="sponsor-image"
            src="ReCodedLogo.jpg"
          />
        </a>
      </div>
      <div
        className="col"
      >
        <a
          href="https://www.re-coded.com/"
        >
          <img
            alt="Re:coded"
            className="sponsor-image"
            src="ReCodedLogo.jpg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
